File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool checkvowel(char c)
    {
        return c=='a' || c=='e' || c=='i' || c=='o' || c=='u' || c=='A' || c=='E' || c=='I' || c=='O' || c=='U';
    }
    string reverseVowels(string s) {
        int i=0,j=s.size()-1;
        while(i<j)
        {
            if(checkvowel(s[i]))
            {
                while(i<j)
                {
                    if(checkvowel(s[j]))
                    {
                        swap(s[i],s[j]);
                        j--;
                        break;
                    }
                    j--;
                }
            }
            i++;
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
   boolean isvowal(char ch)
{
    if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u' || ch=='A' || ch=='E' || ch=='I' || ch=='O' || ch=='U')
        return true;
    return false;
}

public String reverseVowels(String s) {
    
    char[] ch= s.toCharArray();
    int j=ch.length-1;
    int i=0;
    
    while(i<j)
    {
        if(isvowal(ch[i]) && isvowal(ch[j]))
        {
            char temp=ch[i];
            ch[i]=ch[j];
            ch[j]=temp;
            i++;
            j--;
        }
        else if(isvowal(ch[i]) && !isvowal(ch[j]))
            j--;
        else if(!isvowal(ch[i]) && isvowal(ch[j]))
            i++;
        else{
            i++;
            j--;
        }
    }
    
    return new String(ch);
}
}